libvmi/os/linux/memory.c

```diff
@@ -54,25 +55,28 @@ linux_get_taskstruct_addr_from_pid(
     tasks_offset = linux_instance->tasks_offset;
 
     /* First we need a pointer to the initial entry in the tasks list.
-     * Note that this is task_struct->tasks, not the base addr
-     *  of task_struct: task_struct base = $entry - tasks_offset.
+     * In newer versions of Linux, the global "init_task" points to
+     * the actual task struct, not to the linked list entry.
      */
-    next_process = vmi->init_task;
-    list_head = next_process;
+    curr_proc = vmi->init_task;
+    vmi_read_addr_va(vmi, curr_proc + tasks_offset, 0, &curr_entry);
+
+    list_head = curr_entry;
 
     do {
-        vmi_read_32_va(vmi, next_process + pid_offset, 0, (uint32_t*)&task_pid);
```

Comment:
As this line indicates, next_process actually pointed to a task_struct, not to a list_head as claimed by the above (removed) comment.